products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -471,6 +471,39 @@ class CAPIBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
     uint64_t count() { return rules.size(); }
   };
 
+  static llb_rule_run_reason_t
+  convertRunReason(core::Rule::RunReason reason) {
```

Comment:
Maybe ad a commend to both types and do a static cast? They're based on the same raw value.

Reply:
I think I prefer the current approach to maintain consistency with the other enums, so we have a clearer boundary between the C & C++ APIs

Reply:
Agree with @owenv, despite the verbosity, forced casts based on assumptions about the underlying integer value can be dangerous.